File: ruby/railshost/app/controllers/benchmark_controller.rb
class BenchmarkController < ApplicationController
  def warmup
    render plain: "OK"
  end

  def alive
    render plain: "OK"
  end

  def platform
    render plain: "Rails #{Rails.version}/Ruby #{RUBY_VERSION} (#{Rails.env})"
  end

  def login
    response = {
      user: params[:user],
      token: Base64.urlsafe_encode64(params[:pass])
    }
    render json: response
  end
end

File: elixir/phoenixhost/lib/phoenixhost_web/controllers/benchmark_controller.ex
defmodule PhoenixhostWeb.BenchmarkController do
  use PhoenixhostWeb, :controller

  def warmup(conn, _params) do
    text conn, "OK"
  end

  def alive(conn, _params) do
    text conn, "OK"
  end

  def platform(conn, _params) do
    {:ok, version} = :application.get_key(:phoenix, :vsn)
    text conn, "Phoenix #{version}/Elixir #{System.version}/OTP #{System.otp_release} (#{Mix.env})"
  end

  def login(conn, %{"user" => user, "pass" => pass}) do
    response = %{
      user: user,
      token: Base.url_encode64(pass)
    }
    json conn, response
  end
  def login(conn, _), do: json(conn, %{error: "bad"})
end
